File: CS/1844_Replace_All_Digits_with_Characters.cs
public class Solution {
    public string ReplaceDigits(string s) {
        StringBuilder result = new StringBuilder();
        char temp = '0';
        foreach(var c in s){
            if(c >= '0' && c <= '9'){
                result.Append((char)(temp + c -'0'));
            }
            else{
                temp = c;
                result.Append(c);
            }
        }
        
        return result.ToString();
    }
}

File: Cpp/1844_Replace_All_Digits_with_Characters.cpp
class Solution {
public:
    string replaceDigits(string s) {
        for(int i = 1; i < s.size(); i+=2){
            s[i] = s[i-1] + s[i] - '0';
        }
        return s;
        /*
        string result;
        char c;
        for(int i = 0; i < s.size(); i++){
            if(i%2 == 0){
                c = s[i];
            }
            else{
                c = (char) (c + s[i] - '0');
            }
            result += c;
        }
        return result;
        */
    }
};
